go.mod

```diff
@@ -0,0 +1,34 @@
+module github.com/kubernetes-sigs/kube-storage-version-migrator
+
+go 1.12
+
+require (
+	github.com/beorn7/perks v1.0.0 // indirect
+	github.com/google/go-cmp v0.3.1 // indirect
+	github.com/googleapis/gnostic v0.2.0 // indirect
+	github.com/imdario/mergo v0.3.7 // indirect
+	github.com/onsi/ginkgo v1.8.0
+	github.com/prometheus/client_golang v0.9.2
+	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
+	github.com/prometheus/common v0.3.0 // indirect
+	github.com/prometheus/procfs v0.0.0-20190503130316-740c07785007 // indirect
+	github.com/spf13/cobra v0.0.5
+	github.com/spf13/pflag v1.0.5
+	golang.org/x/crypto v0.0.0-20190701094942-4def268fd1a4 // indirect
+	golang.org/x/sys v0.0.0-20191008105621-543471e840be // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	k8s.io/api v0.0.0-kubernetes-1.16.2
+	k8s.io/apiextensions-apiserver v0.0.0-kubernetes-1.16.2
+	k8s.io/apimachinery v0.0.0-kubernetes-1.16.2
+	k8s.io/client-go v0.0.0-kubernetes-1.16.2
+	k8s.io/klog v0.4.0
+	k8s.io/kube-aggregator v0.0.0-kubernetes-1.16.2
+)
+
+replace (
```

Comment:
Is the "replace" section manually edited? Why are the specific revisions used here, instead of v0.0.0-kubernetes-1.16.2?

Reply:
Yes, the `replace` section is manually edited. 

`go.mod` does not allow you to specify a branch or non-semver tag. If you are not using a semver tag (e.g. `v1.0.0`), which kubernetes does not at the moment, you must use a revision, and that is converted to a pseudo-version. Unfortunately, when you run certain `go` commands (such as `go build`), there is a chance that the pseudo version will be replaced with the latest from `master`. So... 
1. I set the `k8s.io/*`  dependencies to a pseudo version of `v0.0.0-kubernetes-1.16.2`. The format accomplishes two things: 1. `go` sees it as a pseudo-version and 2. the `kubernetes-1.16.2` part is a reminder of what I really want to require.
2. Adding `replace` entries for `k8s.io/*` entries accomplishes the following: 1. the psedu-version in the `require` section is ignored, and 2. The revision of the `k8s.io/*` module is effectively _pinned_ and won't change unless explicitly updated.

Reply:
Thanks for the explanation.

If making "v0.0.0-kubernetes-1.16.2" a pseudo version isn't a common practice, can you document the purpose in comments?

In the future, when we need to bump the version of k8s.io/*, I think we need to manually edit the "require" section and the "replace" section, right?

> Unfortunately, when you run certain go commands (such as go build), there is a chance that the pseudo version will be replaced with the latest from master

Can you say what commands exactly?
Anyhow, if we just use the revision based pseudo-version generated by `go`, it's not obvious what tag the revision matches to.

Reply:
Let's do the 1.17 port soon. It has proper semver tags.